packages/devtools_app/lib/src/screens/inspector/layout_explorer/ui/utils.dart

```diff
@@ -284,52 +282,65 @@ class AnimatedLayoutProperties<T extends LayoutProperties>
     return _children;
   }
 
-  List<double> _lerpList(List<double> l1, List<double> l2) {
+  List<double?> _lerpList(List<double?> l1, List<double?> l2) {
```

Comment:
Look at the references for this method. I don't see anywhere that we pass in a List of type `double?`.

Reply:
this should definitely be non-null. In general, having `List<int?>`, List<bool?>` or List<double?>` is a code smell and should probably be a lint.

Reply:
In this case yes. In general, I do not see why such code smells. Some elements may have value not defined. For example, in a column, some elements have height specified and others need to share what is remaining equally.
Am I missing something?

Reply:
Maybe my comment was too extreme. There are a few cases where it can be handy to have a nullable list of a primitive type but in general I think it is a code smell in the context of migrating code. Any time I see a `List<double?>` I'd want a strong reason why it has to be nullable. This is because there are natural sentinel values that often make sense to use (e.g. `0` or `NaN`) rather than null so you really have to have a good reason to make the values nullable.

Searching for uses of List<bool?> and List<int?> across code bases, I see a lot more cases where it indicates the null safety migration could have been done better than cases where List<int?> seems like a reasonable design call. Often the list doesn't really need to be nullable as all content is expected to eventually have a non-null value or there was just a bug in running the migration tool so List<int?> snuck in.


For example, if you have a `List<bool?>` and really need null values, probably defining an enum with 3 values would be cleaner. For the Column case, is using `0` or another marker value for omitted height constraints cleaner?